python/tissue:histone_analysis.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jun 26 09:37:33 2019
+
+@author: marcdimeo
+"""
+
+"""
+This script is intended to serve as test/method to group cell lines according to their tissues and histones
+which will that be used for tissue specific analysis and histone correlation
+"""
+
+
+import pandas as pd
+import numpy as np
+
+"""
+FORMATTING DATA --> TISSUE ANALYSIS
+"""
+
+#Take cell lines and match them with tissues and average out results
+#Cell line --> Tissue correspondence is found in CCLE_GCP.csv\
+#CCLE_GCP is really the h3_relval data????
+
+df = pd.read_csv(r'./../data/CCLE_GCP.csv')
+
+temp_df = pd.DataFrame(df['CellLineName'].str.split('_',1).tolist(),columns=['CellLine', 'Tissue'])
+
+i = 0 
+for tissue in temp_df.Tissue:
+    tissue = tissue.split('_')
+    tissue = " ".join(tissue)
+    temp_df.Tissue[i] = tissue
+    i = i+1
+    
+del df['CellLineName']
+
+df.insert(1, 'CellLine', temp_df.CellLine , True)
+df.insert(2, 'Tissue', temp_df.Tissue, True)
+
+#Fill NaN values
+df = df.fillna(method='ffill')
+
+#Convert values into arrays and lists
+cellline_tissue  = temp_df.values
+
+histones = list(df.columns.values)
+histones = histones[3:]
+
+expression_df = df.copy()
+del expression_df['BroadID']
+del expression_df['CellLine']
+del expression_df['Tissue']
+
+expression_df = expression_df.values
+
+tissue_dict ={}
+for data in cellline_tissue:
+    tissue_dict[data[0]] = data[1]
+
+"""
+AVERAGING OUT EXPRESSION VALUES --> TISSUE ANALYSIS
+"""
+
+lung =[]
+ovary = []
+large_intestine = []
+central_nervous_system = []
+haematopoietic_and_lymphoid_tissue =[]
+pancreas = []
+upper_aerodigestive_tract = []
+breast = []
+prostate =[]
+stomach = []
+endometrium = []
+bone = []
+skin =[]
+liver = []
+fibroblast = []
+soft_tissue = []
+biliary_tract = []
+autonomic_ganglia =[]
+pleura = []
+urinary_tract = []
+kidney = []
+oesophagus = []
+thyroid = []
+salivary_gland = []
+
+i = 0
+for celllines in tissue_dict:
```

Comment:
This needs to be cleaned up so that it's automated. What I would do is save the tissues you extracted from the CCLEdata script into a text file, and read the tissues into this function. Under a for loop, iterate these if, then statements. Same thing for all repetitive code below.